src/test/java/com/github/khanmurtuza/FileCrawler.java

```diff
@@ -0,0 +1,34 @@
+package com.github.khanmurtuza;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+
+/**
+ * Crawl file on local disk
+ */
+public class FileCrawler implements Crawler {
+
+    public Document getDocument(final String link) throws IOException {
+        return Jsoup.parse(new File(link), "UTF-8");
+    }
+
+    public boolean shouldCrawl(final String link) {
+        return true;
```

Comment:
I wanted to add unit test to validate EmailScraper business logic, in this case skipping url which does not match input domain.

For FileCrawler I took a shortcut to assume all links(file reference) are in same domain, but ideally could have added some logic that will skip certain link pattern from test file